File: Java/Gas-Station.java
public class Solution {
    /**
     * @param gas: an array of integers
     * @param cost: an array of integers
     * @return: an integer
     */
    public int canCompleteCircuit(int[] gas, int[] cost) {
        // write your code here
        int index = 0;
        int sum = 0, total = 0;
        for (int i = 0; i < gas.length; i ++) {
            sum += gas[i] - cost[i];
            total += gas[i] - cost[i];
            if (sum < 0) {
                sum = 0;
                index = i + 1;
            }
        }
        if (total < 0) return -1;
        return index;
    }
}

File: Python/Gas-Station.py
class Solution:
    # @param gas, a list of integers
    # @param cost, a list of integers
    # @return an integer
    def canCompleteCircuit(self, gas, cost):
        # write your code here
        res = 0
        cur = 0
        index = 0
        for i in range(len(gas)):
            if cur < 0:
                cur = 0
                index = i
            res += (gas[i] - cost[i])
            cur += (gas[i] - cost[i])
        if res < 0:
            return -1
        return index